Multi node cron jobs.

At the moment if you were to start a neoq worker on 2 different nodes with a single PostgreSQL backend and have crons configured for them. 

Every time a cron would be scheduled, one node will ultimately insert the cron into the jobs table while the other nodes fail with an error `level=ERROR msg="error queueing cron job" error="duplicate job"`.

Is the intended use case that a single node would orchestrate the enqueuing of these crons? Or that the contention and resulting error about duplicate job from the additional nodes is intended?

Reopening

```
=== Failed
=== FAIL: backends/postgres TestMultipleCronNodes (3.18s)
==================
WARNING: DATA RACE
Read at 0x00c0000a687c by goroutine 4624:
  github.com/acaloiaro/neoq/backends/postgres_test.TestMultipleCronNodes()
      /__w/neoq/neoq/backends/postgres/postgres_backend_test.go:432 +0x924
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1576 +0x216
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1629 +0x47

Previous write at 0x00c0000a687c by goroutine 5174:
  sync/atomic.AddInt32()
      /usr/local/go/src/runtime/race_amd64.s:281 +0xb
  sync/atomic.AddUint32()
      <autogenerated>:1 +0x1a
  github.com/acaloiaro/neoq/handler.Exec.func1()
      /__w/neoq/neoq/handler/handler.go:137 +0xe2
  github.com/acaloiaro/neoq/handler.Exec.func2()
      /__w/neoq/neoq/handler/handler.go:138 +0x58

Goroutine 4624 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1629 +0x805
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:2036 +0x8d
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1576 +0x216
  testing.runTests()
      /usr/local/go/src/testing/testing.go:2034 +0x87c
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:1906 +0xb44
  github.com/acaloiaro/neoq/backends/postgres_test.TestMain()
      /__w/neoq/neoq/backends/postgres/postgres_backend_test.go:78 +0x2e
  main.main()
      _testmain.go:94 +0x33d

Goroutine 5174 (finished) created at:
  github.com/acaloiaro/neoq/handler.Exec()
      /__w/neoq/neoq/handler/handler.go:[115](https://github.com/acaloiaro/neoq/actions/runs/6434396784/job/17473540486#step:4:116) +0x309
  github.com/acaloiaro/neoq/backends/postgres.(*PgBackend).handleJob()
      /__w/neoq/neoq/backends/postgres/postgres_backend.go:732 +0x924
  github.com/acaloiaro/neoq/backends/postgres.(*PgBackend).start.func2()
      /__w/neoq/neoq/backends/postgres/postgres_backend.go:549 +0x32f
==================
time=2023-10-06T17:10:09.006Z level=ERROR msg="error enqueueing job" error="unable add job to queue: context canceled"
time=2023-10-06T17:10:09.006Z level=ERROR msg="error queueing cron job" error="error committing transaction: conn closed"
    testing.go:1446: race detected during execution of test

DONE 115 tests, 1 failure in [137](https://github.com/acaloiaro/neoq/actions/runs/6434396784/job/17473540486#step:4:138).274s
make: *** [Makefile:22: test] Error 1
```

--- a/backends/postgres/postgres_backend_test.go
+++ b/backends/postgres/postgres_backend_test.go
@@ -213,7 +213,7 @@ func TestMultipleProcessors(t *testing.T) {
 	wg.Wait()
 
 	// Make sure that we executed the expected number of jobs.
-	if execCount != uint32(ConcurrentWorkers) {
+	if atomic.LoadUint32(&execCount) != uint32(ConcurrentWorkers) {
 		t.Fatalf("mismatch number of executions. Expected: %d Found: %d", ConcurrentWorkers, execCount)
 	}
 }
@@ -429,11 +429,11 @@ func TestMultipleCronNodes(t *testing.T) {
 
 	// allow time for listener to start and for at least one job to process
 	time.Sleep(WaitForJobTime)
-	if jobsCompleted == 0 {
+	if atomic.LoadUint32(&jobsCompleted) == 0 {
 		t.Fatalf("no jobs were completed after %v", WaitForJobTime)
 	}
 
-	if duplicateJobs > 0 {
+	if atomic.LoadUint32(&duplicateJobs) > 0 {
 		t.Fatalf("some jobs were processed more than once")
 	}
 }

